content/posts/2021/reana-client-python-api.md

```diff
@@ -0,0 +1,223 @@
+    parameters=my_inputs,
+    workflow_engine=my_workflow_type)
+```
+
+After creating our workflow we also need to upload all the input files to
+the workflow workspace. Same functionality can be achieved with `reana-client`
+[upload](https://reana-client.readthedocs.io/en/latest/#reana-client-upload)
+command. It can be done as follows:
+
+```python
+from reana_client.api.client import upload_to_server
+
+abs_path_to_input_files = [os.path.abspath(f) for f in my_inputs['files']]
+upload_to_server(my_workflow_name, abs_path_to_input_files, my_reana_token)
+```
+
+At this point we have done all what is needed to start our workflow in REANA.
+To do it with `reana-client` we could call [start](https://reana-client.readthedocs.io/en/latest/#reana-client-start)
+command. But in this case it can be started by simply calling `start_workflow`
+function and defining workflow name and access token:
+
+```python
+from reana_client.api.client import start_workflow
+
+start_workflow(my_workflow_name, my_reana_token, {})
+```
+
+After starting our wokflow we might want to check what is the current status of it.
+To inspect the workflow status using `reana-client` we could use [status](https://reana-client.readthedocs.io/en/latest/#reana-client-status)
+command. Similarly, it can be achieved with `get_workflow_status` function.
+Since it will take couple of seconds for all the computations to be finished
+while executing our workflow, we will use a loop to wait and regurlary check
+what is the current status of our workflow:
+
+```python
+import time
+
+while True:
+    from reana_client.api.client import get_workflow_status
+
+    status_details = get_workflow_status(my_workflow_name, my_reana_token)
+    if status_details['status'] == 'finished':
+        break
+    print('Current status: ', status_details['status'])
+    time.sleep(1)
+```
+
+Once the workflow is finished, we usually want to list all the output files which our workflow
+produced. With `reana-client` it can be achieved by using
+[ls](https://reana-client.readthedocs.io/en/latest/#reana-client-ls)
+command. Similarly, it can be done in the following manner:
+
+```python
+from reana_client.api.client import list_files
+
+list_files(my_workflow_name, my_reana_token)
+```
+
+After listing all the files we may want to download some of the results for inspection.
+With `reana-client` we could use [download](https://reana-client.readthedocs.io/en/latest/#reana-client-download)
+command. In this case to download a binary blob of a final plot which our
+workflow produced all we need to do is to use `download_file` function:
+
+```python
+from reana_client.api.client import download_file
+
+output_filename = 'results/plot.png'
+file_binary_blob = download_file(
+    my_workflow_name, output_filename, my_reana_token)
+```
+
+To display the image we can use Python Imaging Library `Pillow`. If you don't have it
+installed in your machine, here is how you can do it:
+
+```
+$ pip install Pillow
```

Comment:
Maybe move to the top to Installing requirements?  Saying that we shall use this library to display plots...

Also, you can amend to `pip install --user ...` so that we don't have to explain virtual environment business.